LibertyDSNP/mrc
# Provider should be able to submit a message for a delegator#103

# Purpose
**From a provider's perspective:** I should be able to add messages to mrc on behalf of user as an option
**From a delegator's perspective:** An authorized Provider should be able to submit messages on my behalf
**Security perspective:** an unauthorized provider or not permissioned (#20) provider should not be able to submit messages on_behalf

Closes #103 

## Solution
- Extend add message extrinsic to allow for an optional ```on_behalf``` parameter which is the MSA id of actual delegator.
- Run checks to ensure a valid and authorized relationship between provider and delegator.
- Add a TODO comment as placeholder for future checks on permissions delegated to provider.
 
## Change summary
* Updated add extrinsic on messages
* Move some msa related primitives to common primitives for sharing 
* Extend AccountProvider trait to implement get_provider_info ensuring decoupling from MSA pallet
* Unit tests

## Outcomes:
 Design doc (#20) should lead to permission level checks on provider for acting on messages
 Creates issue: #140
## Steps to Verify
1. Two unit tests are added to mock two scenarios where a delegator has authorized provider and also a case where provider is not authorized 
2. Run MRC and verify using extrinsic calls from dashboard
